cloud-platform/add-secrets-to-deployment.md

```diff
@@ -0,0 +1,234 @@
+
+## Prerequisites
+
+This guide assumes the following:
+
+* You have previously set up an env. See [Creating a Cloud Platform Environment](/cloud-platform/env-create)
+* You have previously deployed your application. See [Deploying an application to the Cloud-Platform](/cloud-platform/app-deploy)
+* Check your deployment is running. See [Interacting with the application](/cloud-platform/app-deploy/#interacting-with-the-application)
+
+## Configuring secrets
+
+The following is an example of encoding (configuring) aws access-key credentials in your deployment.
+See [kuberenetes using secrets as environment variables](https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-environment-variables)
+
+for detailed information regarding providing base64 values in secret objects to Kuberenetes pods)
+
+Create your AWS Credentials access key (making a note of the aws_access_key_id and aws_secret_access_key)
+
+### base64-encode your secret as follows:
+
+In this example  aws_access_key_id is 'AKIAFTKSAW15HJLOGD'. Issue the following command to base64-encode:
+
+```
+echo -n 'AKIAFTKSAW15HJLOGD' | base64 -b0
+```
+
+This will return the encoded id 'QUtJQUZUS1NBVzE1SEpMT0dE'
+
+In this example the is aws_secret_access_key 'g8hjpmhvgfhk4547gfdshhjj'. Issue the following command to base64-encode:
+
+```
+echo -n 'QUtJQUZUS1NBVzE1SEpMT0dE' | base64 -b0
+```
+
+This will return the encoded secret 'UVV0SlFVWlVTMU5CVnpFMVNFcE1UMGRF' 
+
+## Creating the secret
+
+There are 2 ways of creating the secret:
+
+1. create a generic secret - for Concourse's git-crypt key
+
+
+2. Create the secret with a YAML file
+
+### 1. create a generic secret - for Concourse's git-crypt key
+
+This is Concourse-specific from a file. With this method output is base64d twice  
+
+put the encoded secret and id into a file (in this example secret-demo.k8s, can be any text file in the format below) as follows:
+
+```
+apiVersion: v1
+
+kind: Secret
+
+metadata:
+
+  name: demosecret
+
+type: Opaque
+
+data:
+
+  aws_access_key_id: QUtJQUZUS1NBVzE1SEpMT0dE
+
+  aws_secret_access_key: UVV0SlFVWlVTMU5CVnpFMVNFcE1UMGRF
+```
+
+create the secret generic secret from that file:
+```
+$kubectl -n demo-app create secret generic demosecret --from-file=demosecret=secret-demo.k8s
+```
+
+'demo-app' being the namespace, 'demosecret' referencing the 'name' in secret-demo.k8s and referencing the file 'secret-demo.k8s'
+this will give the following screen output
+
+
+```
+secret "demosecret" created
+```
+To view the output of your secret:
+
+```
+$ kubectl get secret demosecret -n demo-app -o jsonpath="{.data.aws_secret_key}"
+QUtJQUZUS1NBVzE1SEpMT0dE
+```
+This can then be piped directly into base64 to get the plaintext value:
+
+```
+$ kubectl get secret demosecret -n demo-app -o jsonpath="{.data.aws_secret_key}" | base64 --decode
+AKIAFTKSAW15HJLOGD
+```
+Optional - to see more detail
```

Comment:
I don't think the extra info returned here is particularly useful to users, so I'd drop this and everything down to line 148, `### 2. Create the secret with a YAML file`